src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -259,43 +363,114 @@ def _construct_job_output_state_set(
                     output_states[job_id] = job.output_state(fetched_states[job_id])
                 else:
                     # fetch the current state without updating it
-                    output_states[job_id] = job.output_state({})
-                    # add an error field with the error message from the failed look up
-                    output_states[job_id]["error"] = error_message
+                    output_states[job_id] = job.output_state(no_refresh=True)
+
+            failed_ids = [job_id for job_id in jobs_to_lookup if job_id not in fetched_states]
+            if failed_ids:
+                self.add_errors_to_results(
+                    output_states, failed_ids, OutputStateErrMsg.QUERY_EE2_STATES, error_message
+                )
```

Comment:
this is now more complex than it was, so I'm not sure it is an improvement...

Reply:
I think the `Enum` with all its glory may not have been great as it did add complexity. But I think here it's streamlined because you have a method that takes the output data structure, the id keys, the error message, and also concatenates errors instead of overwriting them. So, `cancel_jobs` calls this method, so the output data structure can accrue both a `ee2.check_jobs` error and a `ee2.cancel_jobs error`.

Also, if you look at python library packages, the code can be really stupid complex. So I may be erring on the side of complexity.